### DISCUSSION TITLE
Gestion des créneaux horaires

### DISCUSSION THREAD

**Discussion by JC144** [JC144/EDF_Simulateur_Prix#27] [Ideas]
Ce soir, je vais essayer d'implémenter la proposition de @zdimension de décrémenter le premier slot pour l'intégrer aux heures de la fin de journée.

On a aussi une bonne proposition de @ver2ter avec un mécanisme pour préciser le pas dans les tarifs, à discuter.
En tout cas, belle remarque pour éviter les erreurs d'arrondis.

Si j'ai encore un peu de temps, je jette un oeil pour gérer les heures creuses à la demi-heure, mais si quelqu'un veut prendre le chantier, pas de soucis.

**Comment by zdimension** [comment]
J'y pense à l'instant, mais pour trouver le pas il y a une solution bête, c'est juste de calculer la différence avec la tranche d'avant. S'il y a 15:00:00 puis 15:15:00, alors on sait que c'est (pour cette tranche-là) 15 minutes. De là, plus qu'à potentiellement multiplier la valeur de conso (e.g. Enedis qui donne des "watt-demi-heure") pour avoir des Wh propres, puis trouver la période tarifaire de la tranche (en fonction du jour et de l'heure de départ) et calculer le prix pour cette tranche.

En somme, pour un jour donné, on n'aurait plus d'"hours" mais des "slices" de durée quelconque, et le prix d'un jour est la somme des slices. C'est l'approche que j'ai dans mon calculateur et je colle à EDF au centime près (sauf pour le passage heure d'été/hiver, mais ça, c'est vraiment désagréable à gérer donc j'ignore)

**Reply by JC144** [reply]
> De là, plus qu'à potentiellement multiplier la valeur de conso (e.g. Enedis qui donne des "watt-demi-heure") pour avoir des Wh propres

Ne plus s'embêter à reconstruire les heures mais juste convertir Watt-demi-heure ou Watt-quart-dheure en Watt-heure et vérifier si c'est dans la tranche ?

Ca semble tout bête, je jetterai un oeil demain à tête reposée.

**Reply by JC144** [reply]
J'ai poussé le fonctionnement par slice, clairement plus élégant. Je vais affiner pour essayer d'avoir Tempo iso puis les heures creuses à la demi-heure.

**Reply by JC144** [reply]
J'ai toujours un léger diff de quelques Wh qui se perdent dans les HC en Tempo sur l'import EDF. Ça a l'air nickel depuis Enedis.

**Reply by zdimension** [reply]
> > De là, plus qu'à potentiellement multiplier la valeur de conso (e.g. Enedis qui donne des "watt-demi-heure") pour avoir des Wh propres
> 
> Ne plus s'embêter à reconstruire les heures mais juste convertir Watt-demi-heure ou Watt-quart-dheure en Watt-heure et vérifier si c'est dans la tranche ?
> 
> Ca semble tout bête, je jetterai un oeil demain à tête reposée.

Oui, ça m'a bcp facilité les calculs. Sur mon calculateur je passe quand même sur des heures à la fin mais uniquement pour la visualisation (le graphique en grille de chaleur), juste en sommant les Wh. Mais derrière, je fais le calcul de prix sur les tranches.

**Reply by zdimension** [reply]
> J'ai toujours un léger diff de quelques Wh qui se perdent dans les HC en Tempo sur l'import EDF. Ça a l'air nickel depuis Enedis.

Pas mon pc sous la main là tout de suite mais : peut être vérifier dans les jours où la couleur change, s'assurer que les tranches de 0h à 5h59 sont bien de la couleur du jour d'avant, ça posait PB précédemment

**Reply by JC144** [reply]
J'avais un doute sur la tranche 6h, mais elle est bien HP alors que c'est les HC qui ont quelques kWh de trop dans mon cas.

**Reply by JC144** [reply]
Bon:
* Enedis : ok
* Léger décalage sur EDF, j'ai toujours pas trouvé l'erreur
* Total:... C'est mieux, mais je ne comprends pas leurs unités et c'est incomparable avec le rapport d'Enedis puisqu'on passe d'un pas à la demi-heure, à un pas à 10 minutes (donc une précision incomparable)

**Comment by Lolo01440** [comment]
Bonjour à tous. et merci pour ce super simulateur. Pas sûr d'être dans la bonne discussion , je doit être très quiche , mais j'ai du mal à comprendre comment remplir le choix des heures Creuse pour le tarif Bleu EDF....qui ne comporte que deux plages HC de 22h30 à 6h30 et HP de 6h30 à 22h30.
Comment rentrer ces heures dans les champs Début de soirée  jour 1 , nuit jour 2  et apers midi jour 2 ?....a quoi correspondent ces 3plages ?

**Reply by JC144** [reply]
C'est une astuce de développeur fainéant 🙄 C'est pas forcément évident de comparer des heures qui sont sur deux jours différents.

Les heures creuses peuvent démarrer en fin de journée, continuer la nuit et parfois il y a d'autres heures creuses dans l'après-midi mais ça dépend des abonnements.

Le premier champ est donc pour la fin de journée, en général de 22h à minuit.
Le second champ pour la nuit, par exemple 00h00-6h.
Et le troisième champ peut être utilisé pour gérer des heures creuses diurnes : 13h00-15h00 par exemple. Mais si vous ne le souhaitez pas, vous pouvez le laisser sur 00h00-00h00.

**Comment by Lolo01440** [comment]
Je suis aussi un développeur fainéant  🙄 Merci pour la rapidité de la réponse, j'avais juste pas percuter que le troisième champs c'était pour déclarer d'éventuelles heures creuses dans la journée, en comparant les résultats avec l'appli EDF j'en avais déduit que les champs par défaut ca marchait très bien pour EDF ( pas de décalage en ayant chipoté en décalant  de 36 min car pour moi les heures pleines commencent à 6h36 !!!)
En tout cas c est un super projet plein de sens, et bravo d'avoir collecté autant de tarifs y compris de fournisseurs que je n'avais jamais entendu parlé !! ;-)
il me reste plous qu'à passer à Tempo et couper du bois pour le poêle pour les jours rouges !!! ( et gagner au bas mots 350 euros par an , associé à petite panneau photovoltaïque monté par moi même ce sera plus de 600 euros d'économies sur la facture EDF

**Reply by JC144** [reply]
La communauté a été folle pour l'ajout des tarifs. Faut que je m'occupe de voir comment les maintenir à jour facilement.
Merci pour le retour en tout cas, ça fait plaisir :)

Tempo se prête super bien avec un poêle en effet. Les jours froids étant souvent dégagés, je regarde pour du solaire d'appoint aussi.
